### PR TITLE
Check for Ledger available status

### PR DESCRIPTION
**Pull request by jacogr**
closes https://github.com/polkadot-js/apps/issues/3788#issuecomment-762499748

Needs https://github.com/polkadot-js/ui/pull/431 (aka Ledger account is always visible under accounts)

@Tbaut Would appreciate a check... still flying "slightly" blind...

### CODE REVIEW

File: packages/react-signer/src/TxSigned.tsx [hunk 1]
```diff
@@ -271,26 +286,29 @@ function TxSigned ({ className, currentItem, requestAddress }: Props): React.Rea
     (): void => {
       setBusy(true);
 
-      setTimeout((): void => {
+      setTimeout(():void => {
```
Comment (jacogr) [inline packages/react-signer/src/TxSigned.tsx:289]:
```suggestion
      setTimeout((): void => {
```

File: packages/react-signer/src/TxSigned.tsx [hunk 2]
```diff
@@ -274,23 +289,26 @@ function TxSigned ({ className, currentItem, requestAddress }: Props): React.Rea
...
-          errorHandler(error as Error);
-        }
+        _unlock()
+          .then((isUnlocked): void => {
+            if (isUnlocked) {
+              isSubmit
+                ? currentItem.payload
+                  ? _onSendPayload(queueSetTxStatus, currentItem, senderInfo)
+                  : _onSend(queueSetTxStatus, currentItem, senderInfo).catch(errorHandler)
+                : _onSign(queueSetTxStatus, currentItem, senderInfo).catch(errorHandler);
```
Comment (jacogr) [inline packages/react-signer/src/TxSigned.tsx:304]: The catch here should be the one below, however I really didn't want to mess around too much with previously working.

**Review by Tbaut — APPROVED** [review]
- Add ledger
- refresh, is still there
- unplug it
- send a tx from it, the browser complains it finds no device 
- plug in with the browser popup still on, the device shows up
- click send again, it shows up on the ledger.

I also tried the same as above, but when plugin it again, I just unlocked it, and did not enter the app, then the signer complains, I entered the app on the ledger and clicked sign again, and it's sent to the ledger as expected.

:100:

**Comment by jacogr** [comment]
@Tbaut Thanks a lot for playing my QA team - this should hopefully sort it itself out by tomorrow (if all goes well). Extra eyes are always great, but it is horrible the untested stuff crosses to your side.

**Pull request merged**